src/pages/managerForm.js

```diff
@@ -1,66 +1,311 @@
 import FooterBar from '@/components/common/FooterBar';
 import styles from '../styles/managerForm.module.css';
+import { firebaseAuth, firebaseDb, firebaseStorage } from "@/firebase/config"
+import { firestore, getFirestore, collection, getDocs, addDoc } from "firebase/firestore"; 
+import { ref, getStorage} from "firebase/storage";
+import firebase from "firebase/app";
+import "firebase/compat/storage";
 
 
-const managerForm = () => {
+function ManagerForm() {
+
+    const [formData, setFormData] = useState({
+        community_name:"",
+        community_manager:"",
+        community_date:"",
+        community_location:"",
+        community_url:"",
+        community_linkedin:"",
+        community_twitter:"",
+        community_facebook:"",
+        community_vision:"",
+        community_description:"",
+    });
+    const [fileUpload, setFileUpload] = useState();
+
+    const router = useRouter()
+
+    const handleChange = (event) => {
+        event.preventDefault();
+        setFormData({ ...formData, [event.target.name]: event.target.value });
+    };
     
     const categoryChoice = () => {
         console.log("clicked the button")
     }
+
+    const db = getFirestore();
+
+    const storage = getStorage();
+
+    const submitCommunity = async (event) => {
+        event.preventDefault();
+            const commCollectionRef = collection(db, "LISTINGS");
+            getDocs(commCollectionRef)
+                .then((snapshot) => {
+                    let listing = []
+                    snapshot.docs.forEach((doc) => {
+                        listing.push({ ...doc.data(), id: doc.id})
+                    })
+                    console.log(listing)
+                })
+                .catch(error => {
+                    console.error(error.message)
+                });
+
+            const docRef = async () => {
+                await addDoc(commCollectionRef, {
+                    community_name: setFormData.community_name.value,
+                    community_manager: setFormData.community_manager.value,
+                    community_date: setFormData.community_date.value,
+                    community_location: setFormData.community_location.value,
+                    community_url: setFormData.community_url.value,
+                    community_linkedin: setFormData.community_linkedin.value,
+                    community_twitter: setFormData.community_twitter.value,
+                    community_facebook: setFormData.community_facebook.value,
+                    community_vision: setFormData.community_vision.value,
+                    community_description: setFormData.community_description.value,
+                })
+                .then(() => {
+                    setFormData.reset("")
+                })
+                    console.log("Document writtern with ID:", docRef.id)
+            }
+
+        } 
+
+    const imageUpload = async (file) => {
```

Comment:
I think you can create a helper file where all these file-related functions can live and then just be imported here. This will make it easier for them to be tested and reuseable.

Reply:
moved helper functions into own file and imported to use in managerForm